packages/grafana-data/src/transformations/transformers/formatString.ts

```diff
@@ -0,0 +1,90 @@
+import { map } from 'rxjs/operators';
+
+import { Field, FieldType } from '../../types';
+import { DataTransformerInfo } from '../../types/transformations';
+
+import { DataTransformerID } from './ids';
+
+export enum FormatStringOutput {
+  UpperCase = 'Upper Case',
+  LowerCase = 'Lower Case',
+  FirstLetter = 'Capitalize First Letter',
+  EveryFirstLetter = 'Capitalize First Letter Of Every Word',
+  PascalCase = 'PascalCase',
+  CamelCase = 'camelCase',
+}
+
+export interface FormatStringTransformerOptions {
+  stringField: string;
+  outputFormat: string;
+}
+
+export const formatStringTransformer: DataTransformerInfo<FormatStringTransformerOptions> = {
+  id: DataTransformerID.formatString,
+  name: 'Format string',
+  description: 'Set the capitalisation of a string field',
+  defaultOptions: { stringField: '', outputFormat: '' },
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        const formatter = createStringFormatter(options.stringField, options.outputFormat);
+
+        if (!Array.isArray(data) || data.length === 0) {
+          return data;
+        }
+
+        return data.map((frame) => ({
+          ...frame,
+          fields: formatter(frame.fields),
+        }));
+      })
+    ),
+};
+
+/**
+ * @internal
+ */
+export const createStringFormatter = (stringField: string, outputFormat: string) => (fields: Field[]) => {
```

Comment:
rather than processing and iterating fields in one functino, can we have

1. based on the output format, get a function (the map function)
2. a loop that will:
  a. find matching fields
  b. replace the values with the processed ones

that will make this a

Reply:
You seem to have sent this prematurely, so to clarify are you suggesting we first find a suitable function based on outputFormat (so only having to retrieve the function once) and THEN search through fields and apply the afforementioned function to matching fields?

Assuming so, that makes sense to me and I'm working on it now.

Reply:
this seems to be resolved